### DISCUSSION TITLE
modify airflow timezone when it deploy in docker

### DISCUSSION THREAD

**Discussion by Slashoper** [apache/airflow#25756] [General]
When I want to modify the  UTC time zone  to  Asia/Shanghai time zone  in docker, what should I do? below is my docker image ,but it's does not work 

`FROM apache/airflow:2.2.4-python3.8
USER root
RUN apt-get update \
  && apt-get install -y --no-install-recommends \
         openjdk-11-jre-headless curl procps\
  && apt-get autoremove -yqq --purge \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
RUN mkdir /opt/beeline \
	&& curl -SL https://archive.apache.org/dist/hadoop/core/hadoop-3.0.0/hadoop-3.0.0.tar.gz \
	| tar -xzC /opt/beeline \
	&& curl -SL https://archive.apache.org/dist/hive/hive-2.1.1/apache-hive-2.1.1-bin.tar.gz \
	| tar -xzC /opt/beeline \
	&& curl -SL https://repo1.maven.org/maven2/io/trino/trino-cli/389/trino-cli-389-executable.jar --output /opt/beeline/trino \
	&& chmod +x /opt/beeline/trino
ENV HADOOP_HOME /opt/beeline/hadoop-3.0.0
ENV HIVE_HOME /opt/beeline/apache-hive-2.1.1-bin
ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
ENV PATH $PATH:$HIVE_HOME/bin
USER airflow
COPY ./config/airflow-kube.cfg /opt/airflow/airflow.cfg
COPY ./config/timezone.py /home/airflow/.local/lib/python3.8/site-packages/airflow/utils/timezone.py
COPY ./config/sqlalchemy.py /home/airflow/.local/lib/python3.8/site-packages/airflow/utils/sqlalchemy.py
RUN pip install --no-cache-dir apache-airflow-providers-apache-spark==2.1.0 \
	&& pip install --no-cache-dir apache-airflow-providers-apache-sqoop==2.1.0 \
	&& pip install trino \
	&& pip install trino[sqlalchemy] \
	&& pip install apache-airflow[password]
`

**Comment by potiuk** [comment]
https://www.google.com/search?q=modify+time+zone+in+docker&oq=modify+time+zone+in+docker&aqs=chrome..69i57j0i10i22i30j0i22i30l2.3352j0j7&sourceid=chrome&ie=UTF-8

**Reply by potiuk** [reply]
Search Google ^^

**Reply by Slashoper** [reply]
Sorry, I didn't express the situation. What I meant was how to modify the time zone deployed by the airflow docker. The default time zone of airflow is UTC time zone

**Reply by potiuk** [reply]
Same. Change zone in your image. Depends how you build it and what kind of image you have - for example if the image is Debian based you need to search how to change it for debian and build yiour image with the time zone you desire. 

Airflow image is a "reference" one - you can extend it as you wish

**Reply by Slashoper** [reply]
thank you for your reply. I did as you said. The time in the container has indeed changed, but the running time of DAG has not changed. The following is a screenshot of the production environment

![image](https://user-images.githubusercontent.com/19834014/186811635-b2225162-5dca-4837-a76c-614de6ab2c64.png)
![image](https://user-images.githubusercontent.com/19834014/186811855-6c031da0-9d91-4b17-9614-291fea0e4991.png)
my docker file add  ENV TZ="Asia/Shanghai"

**Reply by potiuk** [reply]
Did you recreate the dags? Dags with already created schedule will remain as they were - because they are kept in the Database.

**Reply by Slashoper** [reply]
Yes , i do .  I solved this problem and added the time zone in the configuration file

**Reply by ganbaaelmer** [reply]
> Yes , i do . I solved this problem and added the time zone in the configuration file

what configuration file? give us directory path

**Comment by wer860612** [comment]
Hi @Slashoper ,
I have the same problem as you
I want to use the system time to rename my file store in Minio, but the time I print still in UTC timezone
can I see your timezone.py and sqlamy.py to know where I didn't change it?

**Reply by potiuk** [reply]
https://linuxize.com/post/how-to-set-or-change-timezone-in-linux/

**Reply by wer860612** [reply]
Hi @potiuk 
The time I printed was correct. but the airflow logging time was adding 8 hours.

**Comment by iRecursion** [comment]
# .env
```ini
AIRFLOW__CORE__DEFAULT_TIMEZONE=Asia/Shanghai
AIRFLOW__WEBSERVER__DEFAULT_UI_TIMEZONE=Asia/Shanghai
```

# Retrieve TimeZone Before
```shell
default@2a4708cced03:/opt/airflow$ cat /etc/timezone && ls -l /etc/localtime
Etc/UTC
lrwxrwxrwx 1 root root 27 Sep 12 00:00 /etc/localtime -> /usr/share/zoneinfo/Etc/UTC
```

# Dockerfile
```Dockerfile
RUN date
ENV TZ=Asia/Shanghai
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
RUN date
```

# Retrieve TimeZone After
```shell
default@54092dee71fa:/opt/airflow$ cat /etc/timezone && ls -l /etc/localtime
Asia/Shanghai
lrwxrwxrwx 1 root root 33 Sep 27 10:46 /etc/localtime -> /usr/share/zoneinfo/Asia/Shanghai
```

**Reply by ganbaaelmer** [reply]
this one saved me
